proto/backup.proto

```diff
@@ -88,6 +88,28 @@ message BackupRequest {
     uint64 rate_limit = 7;
     // The concurrency for executing the backup request in every tikv node.
     uint32 concurrency = 8;
+
+    oneof storage_backend {
+        Local local = 9;
+        S3 s3 = 10;
+    }
+}
+
+message Local {
+    string path = 1;
```

Comment:
New client should use this `path` for local storage, and TiKV server will try to use `storage_backend` and fallback to the `path` field in `BackRequest`.

Reply:
Given [protobuf's backward-compatibility design](https://developers.google.com/protocol-buffers/docs/proto3#updating), you could turn that `oneof` into

```protobuf
oneof storage_backend {
    string local_path = 6;
    bool noop = 9;
    S3 s3 = 10;
}
```

and drop the `string path = 6` field.

This assumes we entirely abandon `path` and not use it to store something like `s3://$bucket/$prefix`.

Reply:
Oh, I forget there is a `noop` storage backend for testing.
Shall we abandon the `path` entirely?

Reply:
I don't think there's any configuration which applies to Local but not other cloud storage (file permission can be easily `chmod`'ed afterwards) 🙃

Reply:
@kennytm Since we add `Noop noop = 9`, I prefer deprecate `string path = 6`, and add a dedicate `message Local` too.